### PR TITLE
Add run and build nix commands for tooling

### PR DESCRIPTION
**Pull request by giomf**
## Introduced Changes
Adding `nix build` and `nix run` commands for twix, fanta and pepsi.

## How to Test
1. run `nix build .#twix --impure`
2. run `nix run .#twix --impure`

Or try with pepsi or fanta

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -4,63 +4,94 @@
...
+              src = ./.;
+              inherit buildInputs;
+              cargoBuildOptions = x: x ++ [ "-p" manifest.name ];
+              cargoTestOptions = x: x ++ [ "-p" manifest.name ];
+            };
         in
         {
+          packages.twix = mktool ./tools/twix/Cargo.toml;
+          packages.pepsi = mktool ./tools/pepsi/Cargo.toml;
+          packages.fanta = mktool ./tools/fanta/Cargo.toml;
```
Comment (knoellle) [inline flake.nix:68]: We have some more tools which don't see much use, but the `behavior_simulator` is used quite regularly.
It also has some lua dependencies which can be a bit annoying. Perhaps managing them with nix might be easier?

Also, `annotato`, which probably also needs the nixGL treatment. 😉

**Review by knoellle — APPROVED** [review]
Works ~~on my machine~~ in my nix, very nice!

This branch needs a rebase because of #675

File: flake.nix [hunk 2]
```diff
@@ -4,69 +4,113 @@
...
+            {
+              type = "app";
+              program = "${twix_wrapper}/bin/twix-wrapper";
+            };
+
+          # Needed for non-nixos systems
+          apps.annotato =
+            let
+              annotato_wrapper = pkgs.writeShellScriptBin "annotato-wrapper" ''
+                ${pkgs.nixgl.auto.nixGLDefault}/bin/nixGL ${self.packages.${system}.annotato}/bin/annotato
```
[the remark this answers is not in the document]
Comment (giomf) [reply flake.nix:88]: Good catch!
Should we also add it to twix?

[the remark this answers is not in the document]
Comment (knoellle) [reply flake.nix:88]: I think so, yes.
Twix currently doesn't take any command line arguments, but we should keep these wrappers identical or even extract them into a function to ensure consistency.

[the remark this answers is not in the document]
Comment (giomf) [reply flake.nix]: Done. Please have a look.

**Review by knoellle — APPROVED** [review]
LGTM

**Pull request merged**